amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/PollingConfig.java

```diff
@@ -153,4 +154,10 @@ public void validateState(final boolean isMultiStream) {
             }
         }
     }
+
+    private void validateMaxRecords() {
+        if (maxRecords() > 10000) {
```

Comment:
We should use DEFAULT_MAX_RECORDS once again here so it applies everywhere. Otherwise we will have to manually update all these places

Reply:
Also DRY (don't repeat yourself). This function already exists. Move this to a helper function if necessary.

Reply:
Also add tests for these validations. and also stream arn validation if it doesnt exist so it wont break in future.

Reply:
Adding a test to validate maxRecords in the next commit. As for the streamArn, they are validated inS: https://github.com/awslabs/amazon-kinesis-client/blob/master/amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java#L63